plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -971,6 +1027,51 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 		fu_device_add_instance_strup(device, "PUBKEY", guid);
 		if (!fu_device_build_instance_id(device, NULL, "USB", "VID", "PID", "PUBKEY", NULL))
 			return FALSE;
+	} else {
+		/* add specific product info */
+		if (self->running_bank == BANK_MASK_CODE) {
+			g_autofree gchar *ic = NULL;
+			ic = fu_common_strsafe((const gchar *)self->static_ts.mask_project_ic_type,
+					       sizeof(self->static_ts.mask_project_ic_type));
+			fu_device_add_instance_str(device, "IC", ic);
+			if (!fu_device_build_instance_id(device,
+							 NULL,
+							 "USB",
+							 "VID",
+							 "PID",
+							 "IC",
+							 "BONDING",
+							 NULL))
+				return FALSE;
+
+		} else {
+			const gchar *vendor = fwupd_device_get_vendor(FWUPD_DEVICE(device));
+			g_autofree gchar *ic = NULL;
+			g_autofree gchar *guid_vs_ts = NULL;
+			guint16 port_num = (self->dynamic_ts.ss_port_number << 8) +
```

Comment:
move the `port_num` before the `g_autofree` locals. to-be-freed variable should be listed at the end.